src/internal/coredb/projects.go

```diff
@@ -17,13 +17,10 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/pfs"
 )
 
-// ID is the auto-incrementing primary key used for entries in postgres tables.
-type ID uint64
```

Comment:
I think you should bring back a `type ProjectID uint64` and use that to enforce typing of projects identifiers.